src/mat/4C_mat_inelastic_defgrad_factors.cpp

```diff
@@ -2436,6 +2441,18 @@ void Mat::InelasticDefgradTransvIsotropElastViscoplast::evaluate_inverse_inelast
   Core::LinAlg::Matrix<3, 3> CredM(true);
   CredM.multiply_tn(1.0, FredM, FredM, 0.0);
 
+  // check whether we have already evaluated the inverse inelastic deformation gradient for the
+  // given reduced deformation gradient
+  Core::LinAlg::Matrix<3, 3> diff_defgrad{true};
+  diff_defgrad.update(1.0, FredM, -1.0, time_step_quantities_.current_defgrad_[gp_], 0.0);
+  if (diff_defgrad.norm2() < 1.0e-15)
```

Comment:
I am wondering whether exact equality would be more appropriate in the present case. If I understand everything correctly, you really expect the exact same value for the def grad.

Side note: normally you want to scale the tolerance to the magnitude of the values in the deformation gradient.

Reply:
Yes, in principle, I do expect the 2-norm to be exactly 0.0. However, I was not quite sure how exact the storing of the computed defgrad is, as well as the norm computation. Modifying it to `if (diff_defgrad.norm2()) == 0.0` leads to the same results in my simulations, so the procedure is quite exact. Thanks for the hint.

Reply:
Well, you are comparing two copies of the same data, so this better be exactly equal ;)